Move doctrine/persistence to dev dependencies

MidnightDesign opened the pull request:
The dependency is not used in production code, only in DoctrineMongoODMModuleTest\Assets\CustomRepositoryFactory.

MidnightDesign committed:
Move doctrine/persistence to dev dependencies

The dependency is not used in production code, only in DoctrineMongoODMModuleTest\Assets\CustomRepositoryFactory.

driehle commented on composer.json:
```diff
@@ -63,6 +62,7 @@
     },
     "require-dev": {
         "doctrine/coding-standard": "^9.0.0",
+        "doctrine/persistence": "^2.4.0",
```
```suggestion
        "doctrine/persistence": "^2.4.0 || ^3.0.0",
```

driehle reviewed (changes requested):
By moving persistence to require-dev, we will automatically allow persistence v3 for this module, as upstream DoctrineModule allows either v2 or v3. Therefore, could you please change it to `^2.4.0 || ^3.0.0`? We will have to ensure that our unit tests run v3, otherwise this change could be harmful for users.

MidnightDesign committed:
Allow doctrine/persistence 3.x

Co-authored-by: Dennis Riehle <webmaster@riehle-web.com>

driehle committed:
ignore PhpStan error

The parameter cannot be typed here, to ensure compatibility of the test with both persistence v2 and v3. The typed parameters were only introduced with persistence v3.

driehle commented:
I have fixed the issue with PhpStan, which was still present in this PR. Since this PR effectively adds a new feature, i.e. the support for persistence v3, we have to release it as a new minor release, i.e. 4.2.0 and not 4.1.2.

Thank you, @MidnightDesign, for your time and contribution!

MidnightDesign commented:
@driehle Would it be possible to release 4.2? We're waiting for this change. Thanks!

driehle commented:
I am planning to first get #270 merged before releasing 4.2.0. However, I expect to make a release today or tomorrow.
